Enable auto EOL on git repo side, fix some character encoding issues

SteveL-MSFT opened the pull request:
Some of the cleanup work from https://github.com/PowerShell/PowerShell/pull/4761

-  add ending newline to files missing it
- update .gitattributes to enforce autocrlf on all text files (this means some of the files show broader changes because it had CRLF on the repo side, but should just be LF)
- added missing <copyright> opening element
- fix mis-encoded character to apostrophe
- fix copyright symbol where it didn't show up correctly in vscode (although it shows up fine here, it's using the unicode symbol instead of the extended ascii one)

cc @mklement0 @markekraus 
<!--

If you are a PowerShell Team member, please make sure you choose the Reviewer(s) and Assignee for your PR.
If you are not from the PowerShell Team, you can leave the fields blank and the Maintainers will choose them for you. If you are familiar with the team, feel free to mention some Reviewers yourself.

For more information about the roles of Reviewer and Assignee, refer to [CONTRIBUTING.md](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md).

-->

SteveL-MSFT committed:
add ending newline to files missing it using:

awk -F: '$2 ~ / no line terminators/ {print $1}' ~/text-files.txt | xargs -I{} sh -c 'printf "\n" >> "$1"' - {}

SteveL-MSFT committed:
update .gitattributes to enforce autocrlf on all text files

SteveL-MSFT committed:
added <copyright> opening element where it was missing
removed file attribute pointing to wrong filename

SteveL-MSFT committed:
fix mis-encoded character to apostrophe

SteveL-MSFT committed:
replace incorrect encoding of copyright symbol with (c)

markekraus commented:
Just to state the obvious, but, the hashes in the tests will need to be updated for the files which were modified.

SteveL-MSFT committed:
[feature]
updated file hashes in the test

SteveL-MSFT committed:
[feature]
updated testfile hash due to change on EOL
changed tests that were always skipping to pending
resaved TestFileCatalog.txt with LF EOL to be consistent with behavior on AppVeyor

SteveL-MSFT commented:
Not sure why the FileCatalog test is failing on AppVeyor.  Passing on my WS2016 and WS2012R2 boxes.

iSazonov commented on test/powershell/engine/Cdxml/assets/CimTest/CdxmlTest.psd1:
```diff
@@ -1,14 +1,14 @@
+    GUID = '41486F7D-842F-40F1-ACE4-8405F9C2ED9B'
+    Author="Microsoft Corporation"
+    CompanyName="Microsoft Corporation"
+    Copyright="(c) Microsoft Corporation. All rights reserved."
```
Why we use defferent "c"? I see:
-  Copyright (c) 
-  Copyright (C) 
-  Copyright (©)

markekraus replied:
I vaguely recall `(c)` is completely wrong, `(C)` has no legal weight, and that `©` should be used unless the intended process or system cannot parse it (in which case fall back to `(C)`. But I can't find a source.

SteveL-MSFT replied:
IANAL, but doing a search tells me the Berne Convention says none of those symbols are actually needed.  Let me follow up with our legal team to see what they recommend so we have consistency.  Based on prior experience, I don't expect them to get back to me very quickly, so I suggest we have this as a separate issue https://github.com/PowerShell/PowerShell/issues/4917

iSazonov replied:
We can look CoreFX experience.

SteveL-MSFT commented on .gitattributes:
```diff
@@ -1 +1,3 @@
 CHANGELOG.md merge=union
+* text=auto
```
From https://www.git-scm.com/docs/gitattributes:

```none
If you want to ensure that text files that any contributor introduces to the repository have their line endings normalized, you can set the text attribute to "auto" for all files.

*	text=auto
```

The actual conversion of files that have CRLF (or worse mixed EOL) will be part of https://github.com/PowerShell/PowerShell/issues/4910 which we'll do after 6.0.0 final since it touches so many files.

SteveL-MSFT commented:
@TravisEz13 no, I created a separate issue for it https://github.com/PowerShell/PowerShell/issues/4917

daxian-dbw commented:
@SteveL-MSFT quick question, I noticed that some `©` has been changed to `(c)` and some are not (e.g. https://github.com/PowerShell/PowerShell/pull/4912/files?w=1#diff-fd6789ade5008bce347f09499cd96bf5R3). Is that intentional?

SteveL-MSFT commented:
@daxian-dbw I only changed the ones that were using the Unicode version that doesn't show up in VSCode (easy to search for).  I left the ones that use extended ascii.  So intentional.
